ballerina/types.bal

```diff
@@ -0,0 +1,1551 @@
+    # The Tableau authentication header. The value is a credentials token from a Tableau server's response to an authentication request.
+    # The Content-Type and Accept headers should be the mediatype of the request and response except in cases where you want to  [explicitly allow other versions of the resource](https://help.tableau.com/current/api/rest_api/en-us/REST/rest_api_concepts_versions.htm#per_resource_versioning).
+    string? X\-Tableau\-Auth?;
+};
+
+# Proxy server configurations to be used with the HTTP client endpoint.
+public type ProxyConfig record {|
+    # Host name of the proxy server
+    string host = "";
+    # Proxy server port
+    int port = 0;
+    # Proxy server username
+    string userName = "";
+    # Proxy server password
+    @display {label: "", kind: "password"}
+    string password = "";
+|};
+
+public type tableau\.nlp\.lens\.publicrest\.v1\.CreateLensRequest record {
+    string name?;
+    string datasource_id?;
+    string project_id?;
+    string description?;
+    boolean is_feedback_enabled?;
+    tableau\.nlp\.lens\.publicrest\.v1\.FieldDetail[] fields?;
+};
+
+public type tableau\.metricqueryservice\.types\.v1\.TargetValueGoal record {
+    decimal value?;
+};
+
+public type tableau\.metricqueryservice\.types\.v1\.BasicSpecification record {
+    tableau\.metricqueryservice\.types\.v1\.Measure measure?;
+    tableau\.metricqueryservice\.types\.v1\.TimeDimension time_dimension?;
+    tableau\.metricqueryservice\.types\.v1\.Filter[] filters?;
+};
+
+# Represents the Headers record for the operation: MetricQueryService_ListMetrics
+public type MetricQueryService_ListMetricsHeaders record {
+    # The Tableau authentication header. The value is a credentials token from a Tableau server's response to an authentication request.
+    # The Content-Type and Accept headers should be the mediatype of the request and response except in cases where you want to  [explicitly allow other versions of the resource](https://help.tableau.com/current/api/rest_api/en-us/REST/rest_api_concepts_versions.htm#per_resource_versioning).
+    string? X\-Tableau\-Auth?;
+};
+
+# Represents the Headers record for the operation: PulseSubscriptionService_BatchGetSubscriptions
+public type PulseSubscriptionService_BatchGetSubscriptionsHeaders record {
+    # The Tableau authentication header. The value is a credentials token from a Tableau server's response to an authentication request.
+    # The Content-Type and Accept headers should be the mediatype of the request and response except in cases where you want to  [explicitly allow other versions of the resource](https://help.tableau.com/current/api/rest_api/en-us/REST/rest_api_concepts_versions.htm#per_resource_versioning).
+    string? X\-Tableau\-Auth?;
+};
+
+public type tableau\.pulse\.insightsservice\.v1\.Goals record {
+    tableau\.metricqueryservice\.types\.v1\.TargetValueGoal target?;
+};
+
+public type tableau\.pulse\.insightsservice\.v1\.Summary record {
```

Comment:
It seems like the generated record type names are not Ballerina-friendly. We can use the—-use-sanitized-oas flag with the Ballerina OpenAPI command to generate Ballerina-friendly record type names. Since we are using Update 10, this feature is available.

@ovindu-a  Shall we generate client code adding ```--use-sanitized-oas flag?

Reply:
Fixed in [6743a6c](https://github.com/ballerina-platform/module-ballerinax-tableau/pull/2/commits/6743a6cca480d2aa332cf707411543d31930409e)